user_connections.go

```diff
@@ -0,0 +1,161 @@
+package sockchat
+
+import (
+	"log"
+	"sync"
+)
+
+// SockchatUserHandler manages user actions
+type SockchatUserHandler interface {
+	MakeRequest(action string, payload any) error
+	Write(msg SocketMessage)
+	AddConnection(conn *SockChatWS)
+	RemoveConnection(conn *SockChatWS)
+	getActiveConnectionsCount() int
+	getNick() string
+}
+
+// ConnectedUsersPool is responsible for tracking all user handlers
+type ConnectedUsersPool struct {
+	handlers map[string]SockchatUserHandler
+	lock     sync.RWMutex
+}
+
+func NewUserManager() *ConnectedUsersPool {
+	manager := &ConnectedUsersPool{
+		handlers: make(map[string]SockchatUserHandler),
+	}
+	return manager
+}
+
+func (m *ConnectedUsersPool) GetHandler(nick string) (SockchatUserHandler, bool) {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	handler, ok := m.handlers[nick]
+	return handler, ok
+}
+
+func (m *ConnectedUsersPool) AddConnection(conn *SockChatWS, store SockchatChannelStore) SockchatUserHandler {
+	handler, ok := m.GetHandler(conn.Nick)
+	if !ok {
+		handler = NewUserHandler(conn, store)
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		m.handlers[conn.Nick] = handler
+	}
+	handler.AddConnection(conn)
+	return handler
+}
+
+func (m *ConnectedUsersPool) RemoveConnection(conn *SockChatWS) {
+	handler, ok := m.GetHandler(conn.Nick)
+	if !ok {
+		log.Printf("could not drop connection - no handler found for %s", conn.Nick)
+		return
+	}
+
+	if handler.getActiveConnectionsCount() > 1 {
+		handler.RemoveConnection(conn)
+	} else {
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		delete(m.handlers, conn.Nick)
+	}
+}
+
+// UserHandler manages connections of a single connected user
+type UserHandler struct {
+	Nick         string
```

Comment:
```suggestion
	nick         string
```
Nick doesn't have to be exported